myceli/src/handlers.rs

```diff
@@ -1,15 +1,19 @@
-use anyhow::Result;
+use anyhow::{anyhow, Result};
 use local_storage::storage::Storage;
 use messages::{ApplicationAPI, DagInfo, DataProtocol, Message};
 use std::path::PathBuf;
 use std::rc::Rc;
 
-pub fn import_file(path: &str, storage: Rc<Storage>) -> Result<Message> {
-    let root_cid = storage.import_path(&PathBuf::from(path.to_owned()))?;
-    Ok(Message::ApplicationAPI(ApplicationAPI::FileImported {
-        path: path.to_string(),
-        cid: root_cid,
-    }))
+pub fn import_file(path: &str, storage: &mut Rc<Storage>) -> Result<Message> {
```

Comment:
`Rc` not `Arc` ?

Reply:
Updated: that Rc is gone now.

We still have 2 different StorageProvider instances happening, FWIW.